Marlin/pinsDebug.h

```diff
@@ -57,372 +57,598 @@ static bool report_pin_name(int8_t pin) {
   #endif
-  #if PIN_EXISTS(RAMPS_D9)
-    PIN_SAY(RAMPS_D9_PIN);
+  #if defined(SS_PIN) && SS_PIN > -1
+    PIN_SAY(SS_PIN)
   #endif
-  #if PIN_EXISTS(RAMPS_D10)
-    PIN_SAY(RAMPS_D10_PIN);
+  #if defined(STAT_LED_BLUE_PIN) && STAT_LED_BLUE_PIN > -1
+    PIN_SAY(STAT_LED_BLUE_PIN)
   #endif
-  #if PIN_EXISTS(MOSFET_D)
-    PIN_SAY(MOSFET_D_PIN);
+  #if defined(STAT_LED_RED_PIN) && STAT_LED_RED_PIN > -1
+    PIN_SAY(STAT_LED_RED_PIN)
   #endif
-
-  #if PIN_EXISTS(TX_ENABLE)
-    PIN_SAY(TX_ENABLE_PIN);
+  #if defined(STEPPER_RESET_PIN) && STEPPER_RESET_PIN > -1
+    PIN_SAY(STEPPER_RESET_PIN)
+  #endif
+  #if defined(SUICIDE_PIN) && SUICIDE_PIN > -1
+    PIN_SAY(SUICIDE_PIN)
+  #endif
+  #if defined(TC1) && TC1 > -1
+    ANALOG_PIN_SAY(TC1)
+  #endif
+  #if defined(TC2) && TC2 > -1
+    ANALOG_PIN_SAY(TC2)
+  #endif
+  #if defined(TEMP_0_PIN) && TEMP_0_PIN > -1
+    ANALOG_PIN_SAY(TEMP_0_PIN)
+  #endif
+  #if defined(TEMP_1_PIN) && TEMP_1_PIN > -1
+    ANALOG_PIN_SAY(TEMP_1_PIN)
+  #endif
+  #if defined(TEMP_2_PIN) && TEMP_2_PIN > -1
+    ANALOG_PIN_SAY(TEMP_2_PIN)
+  #endif
+  #if defined(TEMP_3_PIN) && TEMP_3_PIN > -1
+    ANALOG_PIN_SAY(TEMP_3_PIN)
+  #endif
+  #if defined(TEMP_4_PIN) && TEMP_4_PIN > -1
+    ANALOG_PIN_SAY(TEMP_4_PIN)
+  #endif
+  #if defined(TEMP_BED_PIN) && TEMP_BED_PIN > -1
+    ANALOG_PIN_SAY(TEMP_BED_PIN)
+  #endif
+  #if defined(TEMP_X_PIN) && TEMP_X_PIN > -1
+    ANALOG_PIN_SAY(TEMP_X_PIN)
+  #endif
+  #if defined(TLC_BLANK_BIT) && TLC_BLANK_BIT > -1
+    PIN_SAY(TLC_BLANK_BIT)
+  #endif
+  #if defined(TLC_BLANK_PIN) && TLC_BLANK_PIN > -1
+    PIN_SAY(TLC_BLANK_PIN)
+  #endif
+  #if defined(TLC_CLOCK_BIT) && TLC_CLOCK_BIT > -1
+    PIN_SAY(TLC_CLOCK_BIT)
+  #endif
+  #if defined(TLC_CLOCK_PIN) && TLC_CLOCK_PIN > -1
+    PIN_SAY(TLC_CLOCK_PIN)
+  #endif
+  #if defined(TLC_DATA_BIT) && TLC_DATA_BIT > -1
+    PIN_SAY(TLC_DATA_BIT)
+  #endif
+  #if defined(TLC_DATA_PIN) && TLC_DATA_PIN > -1
+    PIN_SAY(TLC_DATA_PIN)
+  #endif
+  #if defined(TLC_XLAT_PIN) && TLC_XLAT_PIN > -1
+    PIN_SAY(TLC_XLAT_PIN)
+  #endif
+  #if defined(TX_ENABLE_PIN) && TX_ENABLE_PIN > -1
+    PIN_SAY(TX_ENABLE_PIN)
+  #endif
+  #if defined(UNUSED_PWM) && UNUSED_PWM > -1
+    PIN_SAY(UNUSED_PWM)
+  #endif
+  #if defined(X_ATT_PIN) && X_ATT_PIN > -1
+    PIN_SAY(X_ATT_PIN)
+  #endif
+  #if defined(X_DIR_PIN) && X_DIR_PIN > -1
+    PIN_SAY(X_DIR_PIN)
+  #endif
+  #if defined(X_ENABLE_PIN) && X_ENABLE_PIN > -1
+    PIN_SAY(X_ENABLE_PIN)
+  #endif
+  #if defined(X_MAX_PIN) && X_MAX_PIN > -1
+    PIN_SAY(X_MAX_PIN)
+  #endif
+  #if defined(X_MIN_PIN) && X_MIN_PIN > -1
+    PIN_SAY(X_MIN_PIN)
+  #endif
+  #if defined(X_MS1_PIN) && X_MS1_PIN > -1
+    PIN_SAY(X_MS1_PIN)
+  #endif
+  #if defined(X_MS2_PIN) && X_MS2_PIN > -1
+    PIN_SAY(X_MS2_PIN)
   #endif
-  #if PIN_EXISTS(RX_ENABLE)
-    PIN_SAY(RX_ENABLE_PIN);
+  #if defined(X_STEP_PIN) && X_STEP_PIN > -1
+    PIN_SAY(X_STEP_PIN)
   #endif
+  #if defined(X_STOP_PIN) && X_STOP_PIN > -1
```

Comment:
I disagree about the "STOP" pins.  Reporting that they exist shouldn't be a problem.  Keeping people from writing to them is a must.

The "don't report protected pins" function is already implementd in the M43 command.  It checks if the pin is protected before calling the "report_pin_name" function so the M43 command won't print out anything about the endstops.